src/java/org/apache/cassandra/io/compress/ZstdDictionaryCompressor.java

```diff
@@ -76,9 +80,28 @@ public static ZstdDictionaryCompressor create(Map<String, String> options)
     {
         int level = getOrDefaultCompressionLevel(options);
         validateCompressionLevel(level);
+        validateTrainingParameter(TRAINING_MAX_DICTIONARY_SIZE_PARAMETER_NAME,
+                                  options.getOrDefault(TRAINING_MAX_DICTIONARY_SIZE_PARAMETER_NAME,
+                                                       DEFAULT_TRAINING_MAX_DICTIONARY_SIZE_PARAMETER_VALUE));
+        validateTrainingParameter(TRAINING_MAX_TOTAL_SAMPLE_SIZE_PARAMETER_NAME,
+                                  options.getOrDefault(TRAINING_MAX_TOTAL_SAMPLE_SIZE_PARAMETER_NAME,
+                                                       DEFAULT_TRAINING_MAX_TOTAL_SAMPLE_SIZE_PARAMETER_VALUE));
         return getOrCreate(level, null);
     }
 
+    private static void validateTrainingParameter(String parameterName, String resolvedValue)
```

Comment:
we might probably move this to `IDictionaryCompressor`. If some other dict compressor is introduced, it would be there to reuse in the impl of that compressor.